tests/unit/test_data_plane.py

```diff
@@ -28,6 +28,11 @@
 )
 
 def get_random_metadata():
+    # return a random value for metadata key 
```

Comment:
Adding randomness in tests can lead to flakey failures where things fail sometimes but not always, pass on re-run, etc. Can we just hardcode a metadata key into the test using this? If we care about all these options, we should test each of them.

Reply:
Adding randomness also increases the chance of catching some weird edge cases that no one thought of (e.g. updating a vector that had some specific combination of metadata keys, but with some different combination in the update).

If we get to the state which tests are unstable (which currently doesn't seem to be the case), I agree we can decrease randomness.